tornado/locks.py

```diff
@@ -79,3 +79,46 @@ def _garbage_collect(self):
             self._timeouts = 0
             self._waiters = collections.deque(
                 w for w in self._waiters if not w.done())
+
+
+class Event(object):
+    """An event blocks coroutines until its internal flag is set to True.
+
+    Similar to `threading.Event`.
+    """
+    def __init__(self):
+        self._future = Future()
+
+    def __str__(self):
+        return '<%s %s>' % (
+            self.__class__.__name__, 'set' if self.is_set() else 'clear')
+
+    def is_set(self):
+        """Return ``True`` if the internal flag is true."""
+        return self._future.done()
+
+    def set(self):
+        """Set the internal flag to ``True``. All waiters are awakened.
+
+        Calling `.wait` once the flag is set will not block.
+        """
+        if not self._future.done():
+            self._future.set_result(None)
+
+    def clear(self):
```

Comment:
If anyone is waiting while the event is cleared, they'll never wake up even if the Event is subsequently set. (`threading.Event.clear` is a little fuzzy too - if a `set` is followed by a `clear` some waiters may erroneously return false). This method just seems like a bad idea to me so I'd just get rid of it.